CODE-GUIDELINES.md

```diff
@@ -37,3 +37,59 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Mock before tests, not during tests
+
+When you mock functions for some tests, it is recommended to define the mocks from the `beforeEach` callback inside a `describe`,
+so the environment created by the mocks is clearly defined in the `describe`, and the `test` is only executing the tests:
+
+You also need to reset (or restore) the mocks from the `afterEach` callback, so these mocks are not used for other tests.
```

Comment:
I don't understand why we would need to do that, when we almost never had to. Having vi.resetAllMock() should be enough in the beforeEach.

Reply:
see discussion https://github.com/podman-desktop/podman-desktop/pull/10541